test/src/main/java/org/apache/accumulo/test/util/Wait.java

```diff
@@ -48,4 +50,13 @@ public static boolean waitFor(final Condition condition, final long duration,
     }
     return conditionSatisfied;
   }
+
+  /**
+   * A retry for use in junit tests when testing asynchronous conditions that are expected to
+   * eventually meet the condition or fail the test. Using this method should be used instead of an
+   * arbitrary sleep and hoping to catch the condition in the expected state.
+   */
+  public static void assertTrueWithRetry(final Wait.Condition condition) throws Exception {
+    assertTrue(waitFor(condition));
+  }
```

Comment:
Using this method isn't any better than not using it. I don't think this is really adding much value here.

```java
  assertTrueWithRetry(x);
  assertTrue(waitFor(x));
```

Also, it seems like somebody was trying to avoid adding JUnit assertions into this class directly, otherwise, I'd expect the waitFor methods to already do this assertion... because everywhere we use it, we already assertTrue on the returned value... except in a few places where it looks like we forgot to check the return value on the assumption that something gets thrown if the timeout is exceeded... but that's not true.

So, if we want to do anything here in this Wait class, it'd be better to just bake in the assertTrue at the end of the existing waitFor method.

Reply:
Originally @ctubbsii I thought the suggestion in PR #3384 to use a static method in the test to replace the assertTrue(waitFor(....)

```
The method would look like:
  private static void waitForTrue(Wait.Condition condition) throws Exception {
    // could specify 20 seconds, but the default is 30, and that's fine. The default delay is 1.
    assertTrue(Wait.waitFor(condition));
  }
```
So that it could be consistent across tests, I though the Wait Util might me a good home.  I though the original `assertTrue(waitFor(x))` was clear enough, but tried to be accommodating to your request.  Maybe I didn't understand the original request?  At this point I do not know what you want.

It seems to be remove the method from Wait to eliminate the junit dependency and just use `assertTrue(waifFor(` in the tests? as it was originally?

Reply:
I guess I have few thoughts here that superseded my previous comment:

1. My suggestion was mainly about reducing boilerplate. Without the custom timeouts, there's already a lot less boilerplate and the method might not be needed.
2. Your name is longer than the one I suggested, making it less beneficial in my mind.
3. My original suggestion was to create a local method in the test, not in the Wait test utility class, because there seemed to be a convention to avoid using the assertions in the Wait class, for some reason. I'm not opposed to putting the assertions directly in the Wait class, but it wasn't done before. If you're willing to put the assertions in the Wait class, then it could be done with the existing methods.
4. After looking more closely at the Wait class, and how it is used, I think all uses of it would benefit from having the assertions checked inside its waitFor methods. If that's done (can be a separate PR), then adding this new method becomes completely unnecessary... it's merely checking the same condition that all the other waitFor uses are... the only difference is that this new method checks it inside the Wait class, and all the other uses check it just outside... it seems kind of pointless to add a method that does the same thing as the existing methods, but is used slightly differently. It's like having a methods for commutative operations like: `multiplyAB(A, B)` and a separate `multiplyBA(B, A)`. We only need the one.

Reply:
FWIW, that change can be done in a separate PR. For this PR, I think you should just use the existing methods, `assertTrue(waitFor(x));` to wrap this up.

Reply:
Removed from Wait util in 55265e8199